repository/api/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/Repository.java

```diff
@@ -20,8 +20,16 @@
 import java.util.Optional;
 
 public interface Repository<T, ID> {
+  /**
+   * @deprecated Use {link {@link #saveSpecification(Object)}} or {link {@link #saveStatus(Object)}}
+   * to save the entity specification or entity status independently.
+   */
   T save(T entity);
 
+  T saveSpecification(T entity);
+
+  T saveStatus(T entity);
+
   Optional<T> findById(ID id);
 
   List<T> findAll();
```

Comment:
Finally handling a TODO from long ago, deprecating the `save` method in favor of two separate `saveSpecification` and `saveStatus` methods.